sources/subs/SettingsFormAdapterFile.class.php

```diff
@@ -0,0 +1,491 @@
+<?php
+
+/**
+ * This class handles display, edit, save, of forum settings.
+ *
+ * @name      ElkArte Forum
+ * @copyright ElkArte Forum contributors
+ * @license   BSD http://opensource.org/licenses/BSD-3-Clause
+ *
+ * This file contains code covered by:
+ * copyright:    2011 Simple Machines (http://www.simplemachines.org)
+ * license:    BSD, See included LICENSE.TXT for terms and conditions.
+ *
+ * @version 1.1 beta 2
+ *
+ */
+class SettingsFormAdapterFile extends SettingsFormAdapterDb
+{
+	/**
+	 * @var int
+	 */
+	private $last_settings_change;
+
+	/**
+	 * @var array
+	 */
+	private $settingsArray = array();
+
+	/**
+	 * @var array
+	 */
+	private $new_settings = array();
+
+	/**
+	 * Helper method, it sets up the context for the settings which will be saved
+	 * to the settings.php file
+	 *
+	 * What it does:
+	 * - The basic usage of the six numbered key fields are
```

Comment:
While you are at it, could it be worth creating 6 named constants like it has been done in other places?
Of course, in most of the cases we will still use the "natural" php filling, but having them could be helpful, even only for documentation purposes.

Something like:
```php
const VARNAME = 0;
const LABEL = 1;
const SAVETO = 2;
const DATATYPE = 3;
const DATASIZE = 4;
const HELP = 5;
```